src/main/java/in/projecteka/consentmanager/user/TransactionPinService.java

```diff
@@ -0,0 +1,24 @@
+package in.projecteka.consentmanager.user;
+
+import in.projecteka.consentmanager.user.model.TransactionPin;
+import lombok.AllArgsConstructor;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import reactor.core.publisher.Mono;
+
+import java.util.Optional;
+
+@AllArgsConstructor
+public class TransactionPinService {
+    private TransactionPinRepository transactionPinRepository;
+    private BCryptPasswordEncoder encoder;
+
+    public Mono<Void> createPinFor(String patientId, String pin) {
+        String encodedPin = encoder.encode(pin);
+        TransactionPin transactionPin = TransactionPin.builder().pin(encodedPin).patientId(patientId).build();
+        return transactionPinRepository.insert(transactionPin);
```

Comment:
Validate whether user has pin already? if not create, otherwise return bad request/ something.